### PR TITLE
Feat: ADD reqwest proxy config.

### PR DESCRIPTION
**Pull request by inszee**
reqwest client can set a proxy for ChatGpt struct.

### CODE REVIEW

**Comment by Maxuss** [comment]
Could you pull latest changes from upstream?

File: src/client.rs [hunk 1]
```diff
@@ -46,6 +53,26 @@ impl ChatGPT {
         Ok(Self { client, config })
     }
 
+    /// Constructs a new ChatGPT API client with provided API Key and Configuration
```
Comment (Maxuss) [inline src/client.rs:56]: Update docs to something like:
```suggestion
    /// Constructs a new ChatGPT API client with provided API Key, Configuration and Reqwest proxy
```

File: src/client.rs [hunk 2]
```diff
@@ -46,6 +53,26 @@ impl ChatGPT {
...
+    pub fn new_with_proxy_andconfig<S: Into<String>>(
```
Comment (Maxuss) [inline src/client.rs:57]: Maybe the name could be changed to something simpler, like:
```suggestion
    pub fn new_with_config_proxy<S: Into<String>>(
```

File: src/client.rs [hunk 3]
```diff
@@ -29,6 +32,10 @@ impl ChatGPT {
         Self::new_with_config(api_key, ModelConfiguration::default())
     }
 
+     /// Constructs a new ChatGPT API client with provided API key and default configuration
+     pub fn new_with_proxy<S: Into<String>>(api_key: S,proxy: Proxy) -> crate::Result<Self> {
+        Self::new_with_proxy_andconfig(api_key, ModelConfiguration::default(),proxy)
```
Comment (Maxuss) [inline src/client.rs:37]: See above about naming change:
```suggestion
        Self::new_with_config_proxy(api_key, ModelConfiguration::default(), proxy)
```

**Review by Maxuss — CHANGES_REQUESTED** [review]
A few changes and it's ready for merge

**Review by inszee — COMMENTED** [review]
good ~

**Review by Maxuss — APPROVED** [review]

**Comment by Maxuss** [comment]
Thank you!

**Pull request merged**